lib/Transforms/Scalar/Scalarizer.cpp

```diff
@@ -290,6 +293,10 @@ bool Scalarizer::doInitialization(Module &M) {
 }
 
 bool Scalarizer::runOnFunction(Function &F) {
+  Module *M = F.getParent();
+  if (M->HasDxilModule() && M->GetDxilModule().GetShaderModel()->IsSM69Plus())
```

Comment:
I think the independent lookup of DXIL version would help here, since it won't do something different when there's no created `DxilModule` already present.

Reply:
I could also replace it with getorcreate. I was a bit nervous about that becuase of how it failed for hlmodules, but I've since learned that it's a lot more robust and I've made the hlmodule variant similarly so.